packages/react/src/combobox/ComboboxAsync.tsx

```diff
@@ -66,145 +55,64 @@ export function ComboboxAsync<Option extends DefaultComboboxOption>({
 	loadOptions: loadOptionsProp,
 	clearable = false,
 	showDropdownTrigger = true,
-	openDropdownOnFocus = true,
 	inputRef: inputRefProp,
 	...props
 }: ComboboxAsyncProps<Option>) {
 	const inputId = useComboboxInputId(id);
 
-	const [state, setState] = useState<{
```

Comment:
This logic has been moved into `useAsync` as it is duplicated across `ComboboxAsync` and `ComboboxAsyncMulti`

I've also refactored this to use `useReducer` instead.